spec-tests/src/test/java/org/candlepin/spec/consumers/ConsumerResourceDevSpecTest.java

```diff
@@ -0,0 +1,348 @@
+            .addAttributesItem(ProductAttributes.ExpiresAfter.withValue("60"))
+            .providedProducts(Set.of(providedProduct1, providedProduct2)));
+
+        ConsumerDTO consumer = userClient.consumers().createConsumer(Consumers.random(owner)
+            .facts(Map.of("dev_sku", devProduct1.getId())));
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+
+        Set<ConsumerInstalledProductDTO> installed = Set.of(installedProductOf(providedProduct1),
+            installedProductOf(providedProduct2));
+
+        consumerClient.consumers().updateConsumer(consumer.getUuid(), new ConsumerDTO()
+            .installedProducts(installed));
+        autoAttachAndVerifyDevProduct(consumer, consumerClient, devProduct1.getId(), null);
+
+        ProductDTO providedProduct3 = hostedTestApi.createProduct(Products.randomEng());
+        ProductDTO upstreamDevProduct = hostedTestApi.updateProduct(devProduct1.getId(), new ProductDTO()
+            .name("updated dev product 1")
+            .addAttributesItem(new AttributeDTO().name("test_attribute").value("test_value"))
+            .addAttributesItem(ProductAttributes.ExpiresAfter.withValue("90"))
+            .providedProducts(Set.of(providedProduct1, providedProduct2, providedProduct3)));
+
+        ProductDTO existingDevProduct = client.ownerProducts().getProductByOwner(owner.getKey(),
+            devProduct1.getId());
+        AsyncJobStatusDTO job = client.owners().refreshPools(owner.getKey(), false);
+        job = client.jobs().waitForJob(job.getId());
+        assertThatJob(job).isFinished();
+
+        ProductDTO updatedDevProduct = client.ownerProducts().getProductByOwner(owner.getKey(),
+            devProduct1.getId());
+
+        // Verify base state
+        assertThat(existingDevProduct)
+            .returns(devProduct1.getId(), ProductDTO::getId)
+            .returns(devProduct1.getName(), ProductDTO::getName)
+            .returns(getAttributeValue(devProduct1, ProductAttributes.ExpiresAfter.key()), x ->
+                getAttributeValue(x, ProductAttributes.ExpiresAfter.key()))
+            .returns(null, x -> getAttributeValue(x, "test_attribute"));
+
+        // Verify updated state
+        assertThat(updatedDevProduct)
+            .returns(devProduct1.getId(), ProductDTO::getId)
+            .doesNotReturn(devProduct1.getName(), ProductDTO::getName)
+            .doesNotReturn(getAttributeValue(devProduct1, ProductAttributes.ExpiresAfter.key()), x ->
+                getAttributeValue(x, ProductAttributes.ExpiresAfter.key()))
+            .returns("test_value", x -> getAttributeValue(x, "test_attribute"));
+
+        assertThat(upstreamDevProduct)
+            .returns(updatedDevProduct.getName(), ProductDTO::getName)
+            .returns(getAttributeValue(updatedDevProduct, ProductAttributes.ExpiresAfter.key()), x ->
+            getAttributeValue(x, ProductAttributes.ExpiresAfter.key()))
```

Comment:
Are we not permitted to indent these carry-on lines without fighting with the linter?

Reply:
I tried. I don't know why these two lines were flagged while the ones in the same method preceding it were not.

Reply:
Figured as much. Our linter doesn't gracefully handle spillover lines when we're already indented for a chained statement. Unfortunate.